packages/components/src/components/SquiggleViewer/VariableBox.tsx

```diff
@@ -22,58 +22,87 @@ type SettingsMenuParams = {
 type VariableBoxProps = {
   value: SqValue;
   heading: string;
-  preview?: React.ReactNode;
-  renderSettingsMenu?: (params: SettingsMenuParams) => React.ReactNode;
-  children: (settings: MergedItemSettings) => React.ReactNode;
+  preview?: ReactNode;
+  renderSettingsMenu?: (params: SettingsMenuParams) => ReactNode;
+  children: (settings: MergedItemSettings) => ReactNode;
 };
 
-export const SqTypeWithCount = ({
-  type,
-  count,
-}: {
+export const SqTypeWithCount: FC<{
   type: string;
   count: number;
-}) => (
+}> = ({ type, count }) => (
   <div className="text-sm text-stone-400 font-mono">
     {type}
     <span className="ml-0.5">{count}</span>
   </div>
 );
 
-export const VariableBox: React.FC<VariableBoxProps> = ({
-  value: { location },
+export const VariableBox: FC<VariableBoxProps> = ({
+  value,
   heading = "Error",
   preview,
   renderSettingsMenu,
   children,
 }) => {
   const setSettings = useSetSettings();
   const focus = useFocus();
-  const { getSettings, getMergedSettings } = useViewerContext();
+  const { editor, getSettings, getMergedSettings, dispatch } =
+    useViewerContext();
+
+  const findInEditor = () => {
+    const locationR = value.path?.findLocation();
+    if (!locationR?.ok) {
+      return;
+    }
+    editor?.scrollTo(locationR.value.start.offset);
+  };
 
   // Since `ViewerContext` doesn't store settings, `VariableBox` won't rerender when `setSettings` is called.
   // So we use `forceUpdate` to force rerendering.
   const [, forceUpdate] = useReducer((x) => x + 1, 0);
 
-  if (!location) {
-    throw new Error("Can't display a locationless value");
+  const { path } = value;
+
+  if (!path) {
+    throw new Error("Can't display a pathless value");
   }
 
-  const settings = getSettings(location);
+  const settings = getSettings(path);
 
   const setSettingsAndUpdate = (newSettings: LocalItemSettings) => {
-    setSettings(location, newSettings);
+    setSettings(path, newSettings);
     forceUpdate();
   };
 
   const toggleCollapsed = () => {
     setSettingsAndUpdate({ ...settings, collapsed: !settings.collapsed });
   };
 
-  const name = locationToShortName(location);
+  const name = pathToShortName(path);
+
+  const saveRef = (element: HTMLDivElement) => {
+    dispatch({
+      type: "REGISTER_ITEM_HANDLE",
```

Comment:
"ITEM" is really generic. Maybe "REGISTER_VIEWER_ITEM_HANDLE?" Or, we could call the viewer element something more unique, like, explorer/visualizer/presenter/variableDisplay.

Reply:
`dispatch` is ViewerContext-specific, so there's no chance for collisions.
Maybe it could `REGISTER_VARIABLE_HANDLE` instead.

Reply:
That seems preferable to me.